collectors/python.d.plugin/am2320/am320.chart.py

```diff
@@ -0,0 +1,60 @@
+try:
+    import board
+    import busio
+    import adafruit_am2320
+    HAS_STUFF = True
+except ImportError:
+    HAS_STUFF = False
+
+
+#from bases.FrameworkServices.SimpleService import SimpleService
+from base import SimpleService
+
+
+ORDER = [
+    'temperature',
+    'humidity',
+]
+
+CHARTS = {
+    'temperature': {
+        'options': [None, 'Temperature', 'celsius', 'temperature', 'am2320.temperature', 'line'],
+        'lines': [
+            ['temperature']
+        ]
+    },
+    'humidity': {
+        'options': [None, 'Relative Humidity', 'percentage', 'humidity', 'am2320.humidity', 'line'],
+        'lines': [
+            ['humidity']
+        ]
+    }
+}
+
+
+class Service(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.am = None
+
+    def check(self):
+        if not HAS_STUFF:
+            self.error("following libraries are needed to get module working: board, busio, adafruit_am2320")
+            return False
+
+        try:
+            i2c = busio.I2C(board.SCL, board.SDA)
+            self.am = adafruit_am2320.AM2320(i2c)
+        except Exception as error:
```

Comment:
> https://github.com/adafruit/Adafruit_CircuitPython_AM2320/blob/26c73be5d524517e3f019716d952004a8f4c4dcb/adafruit_am2320.py#L87-L123


> adafruit_am2320.AM2320(i2c)

__init__:
 - `ValueError`

temperature/humidity:
 - `OSError`
 - `RuntimeError`

Since they dont use library specifif exceptions probably it is ok to use `Exception` here. 

What do you think @Ferroin ?

Reply:
I still think we should check those specific errors instead of a blanket `Exception` case.  I'll open an issue in the upstream repo to hopefully get the case of them raising a `RuntimeError` instead of using  a proper local exception class fixed.

Reply:
@tommybuck

use `ValueError` here